src/mscorlib/src/System/Collections/Generic/List.cs

```diff
@@ -208,10 +208,36 @@ public class List<T> : IList<T>, System.Collections.IList, IReadOnlyList<T>
         // increased by one. If required, the capacity of the list is doubled
         // before adding the new element.
         //
-        public void Add(T item) {
-            if (_size == _items.Length) EnsureCapacity(_size + 1);
-            _items[_size++] = item;
+        public void Add(T item)
+        {
+            if (_size < _items.Length)
+            {
+                AddWithoutResize(item);
+            }
+            else
+            {
+                AddWithResize(item);
+            }
+        }
+
+        // Separated out of List.Add to improve inlinability of both functions
+        private void AddWithoutResize(T item)
```

Comment:
Updated the asm from keeping them in same method with aggressive inline; is reduced by 10 bytes; with potential for the range check to be eliminated.